pkg/storage/limits/limits.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package limits
+
+import "fmt"
+
+// Any storage is limited, so we have to be sure that data we are going to store would fit underlaying storage structures
+// We could delegate this job to a specific storage plugin, but in this case it has to know too much about job/test descriptor etc
+// Another approach is just fail on insertion of invalid data, but we have to try our best in prevention of this situation
+// (Image a test failing after multiple hours with a error like "you step label is too long!") So as a tradoff we will try
+// to establish and enforce some system wide limits, and make storage plugin developer's life easier
+
+// LimitsValidator provides methods to validate data size from storage perspective
+type LimitsValidator interface {
+	ValidateTestName(testName string) error
+	ValidateTestStepLabel(testStepLabel string) error
+	ValidateEventName(eventName string) error
+	ValidateReporterName(reporterName string) error
+	ValidateJobName(jobName string) error
+	ValidateRequestorName(jobRequestor string) error
+	ValidateServerID(serverID string) error
+
+	// Target name/id??
+}
+
+// ErrParameterIsTooLong implements "error" for generic length check error.
+type ErrParameterIsTooLong struct {
+	DataName  string
+	MaxLen    int
+	ActualLen int
+}
+
+func (err ErrParameterIsTooLong) Error() string {
+	return fmt.Sprintf("%s is too long: %d > %d", err.DataName, err.ActualLen, err.MaxLen)
+}
+
+// Validator is an instance of current storage limits validator
+var Validator LimitsValidator = &BasicStorageLimitsValidator{}
+
+// BasicStorageLimitsValidator is a simple LimitsValidator implementation
+type BasicStorageLimitsValidator struct{}
```

Comment:
what makes it "basic" ? Do we expect other implementations other than for testing?

Reply:
I just wanted some name for the interface implementation. I was thinking to call the interface ILimitsValidator or LimitsValidatorInterface, but decided that's better to add something to the implementation name and didn't come up with anything better than "basic":))

Reply:
Out of curiosity, why do we need an interface? Just in case: https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Reply:
Would testing be impacted if we move away from an interface?

Reply:
I'm not familiar with interface usage patterns in go, sorry:) I was thinking that by using interface we could easily implement multiple validators, or say make storage plugin to return its own implementation, thus make the limits storage plugin dependant. There are many modules which could call the validator, so I thought that it's not very good to declare bunch of interfaces.. The idea was to keep everything in one place..

Reply:
I am in favour of multiple implementations, simply because there could be multiple storage backends (where a custom plugin will help) and backend configurations (where a parametrized plugin will help)

Reply:
Removed the interface. Probably it's better to define an interface from storage plugin to validator in future.